processor/logdeduplicationprocessor/processor.go

```diff
@@ -0,0 +1,125 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logdeduplicationprocessor
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+// logDedupProcessor is a logDedupProcessor that counts duplicate instances of logs.
+type logDedupProcessor struct {
+	emitInterval time.Duration
+	aggregator   *logAggregator
+	consumer     consumer.Logs
+	logger       *zap.Logger
+	cancel       context.CancelFunc
+	wg           sync.WaitGroup
+	mux          sync.Mutex
+}
+
+func newProcessor(cfg *Config, consumer consumer.Logs, logger *zap.Logger) (*logDedupProcessor, error) {
+	// This should not happen due to config validation but we check anyways.
+	timezone, err := time.LoadLocation(cfg.Timezone)
+	if err != nil {
+		return nil, fmt.Errorf("invalid timezone: %w", err)
+	}
+
+	return &logDedupProcessor{
+		emitInterval: cfg.Interval,
+		aggregator:   newLogAggregator(cfg.LogCountAttribute, timezone),
+		consumer:     consumer,
+		logger:       logger,
+	}, nil
+}
+
+// Start starts the processor.
+func (p *logDedupProcessor) Start(_ context.Context, _ component.Host) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	p.cancel = cancel
+
+	p.wg.Add(1)
+	go p.handleExportInterval(ctx)
+
+	return nil
+}
+
+// Capabilities returns the consumer's capabilities.
+func (p *logDedupProcessor) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: true}
+}
+
+// Shutdown stops the processor.
+func (p *logDedupProcessor) Shutdown(_ context.Context) error {
+	p.cancel()
+	p.wg.Wait()
+	return nil
+}
```

Comment:
Should wait on the passed in context here, in case p.wg.Wait() doesn't return in a timely matter?

Reply:
Yeah, I missed this when rewriting it good call.